src/components/Trade/hooks/useTradeRoutes/useTradeRoutes.ts

```diff
@@ -23,7 +24,9 @@ export const useTradeRoutes = (
   const { updateQuote, getDefaultPair, swapperManager } = useSwapper()
   const buyTradeAsset = getValues('buyAsset')
   const sellTradeAsset = getValues('sellAsset')
-  const feeAssetId = chainIdToFeeAssetId(sellTradeAsset?.asset?.chainId ?? 'eip155:1')
+  const feeAssetId = getChainAdapters()
+    .get(sellTradeAsset?.asset?.chainId ?? 'eip155:1')
+    ?.getFeeAssetId() as AssetId
```

Comment:
Oh how I'd love to kill the casting and use explicit types (`const feeAssetId: AssetId`), but if we are using a `Map` object it can in theory be `undefined`. 

We could look to wrap `ChainAdapterManager` in getters and setters to fix this, but that's for another day.

Reply:
Yeah there are two reasons for that:

- The first one is because the map is typed as a `Map<ChainId, ChainAdapter<ChainId>` but both the `sellTradeAsset?.asset?.chainId` and `eip155:1` accessors are strings. Now that `eip155:1` is replaced by `ethChainId` (a `ChainId`), the only thing remaining to fix the accessors is to fix the `Asset` type so that `assetId` and `chainId` are `AssetId` and `ChainId` vs. strings currently - authored https://github.com/shapeshift/lib/pull/893 for it
- The second one is that `getChainAdapters()` can be undefined, that's by design with how that `getOrElse()` singleton of sorts work. Casting `getChainAdapters()` to `ChainAdapterManager` should do the trick for now (given that we update the `Asset` types)

Updated in fdf566cd4117707ce5dcf39ec6bca251abd90be8

Reply:
Excellent assessment, thank you ser.

I'll monitor the progress of https://github.com/shapeshift/lib/pull/893!

Reply:
Here we go: https://github.com/shapeshift/web/pull/2228 no more casting, only a `!`